tests/test_integration.py

```diff
@@ -0,0 +1,540 @@
+test2  .* 'pkg1==2.0' 'pkg2==4.0'
+""".lstrip(),
+        result.stdout,
+    )
+    assert result.returncode == 0
+
+
+def test_run(tmp_path: pathlib.Path, tmp_run: _T_TmpRun) -> None:
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    pys=[3],
+    pkgs={
+        "pytest": [""],
+    },
+    venvs=[
+        Venv(
+            name="pass",
+            command="pytest test_success.py",
+        ),
+        Venv(
+            name="fail",
+            command="pytest test_failure.py",
+        ),
+    ],
+)
+""",
+    )
+    success_path = tmp_path / "test_success.py"
+    success_path.write_text(
+        """
+def test_success():
+    assert 1 == 1
+""",
+    )
+    fail_path = tmp_path / "test_failure.py"
+    fail_path.write_text(
+        """
+def test_failure():
+    assert 1 == 0
+""",
+    )
+    result = tmp_run("riot run -s pass")
+    print(result.stdout)
+    assert re.search(
+        r"""
+============================= test session starts ==============================
+platform.*
+rootdir:.*
+collected 1 item
+
+test_success.py .*
+
+============================== 1 passed in .*s ===============================
+
+-------------------summary-------------------
+✓ pass: .*
+1 passed with 0 warnings, 0 failed\n""".lstrip(),
+        result.stdout,
+    )
+    assert result.stderr == ""
+    assert result.returncode == 0
+
+    result = tmp_run("riot run -s fail")
+    assert "x fail:  pythonInterpreter(_hint='3') 'pytest'\n" in result.stdout
+    assert result.stderr == ""
+    assert result.returncode == 1
+
+
+def test_run_cmdargs(tmp_path: pathlib.Path, tmp_run: _T_TmpRun) -> None:
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    pys=[3],
+    name="test_cmdargs",
+    command="echo hi",
+)
+""",
+    )
+    result = tmp_run("riot run -s test_cmdargs -- -k filter")
+    assert "cmdargs=-k filter" not in result.stdout
+    assert result.stderr == ""
+    assert result.returncode == 0
+
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    pys=3,
+    name="test_cmdargs",
+    command="echo cmdargs={cmdargs}",
+)
+""",
+    )
+    result = tmp_run("riot run -s test_cmdargs -- -k filter")
+    assert "cmdargs=-k filter" in result.stdout
+    assert result.stderr == ""
+    assert result.returncode == 0
+
+
+def test_dev_install_fail(tmp_path: pathlib.Path, tmp_run: _T_TmpRun) -> None:
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    pys=3,
+    name="test",
+    command="echo hello",
+)
+""",
+    )
+    result = tmp_run("riot run test")
+    assert 'ERROR: File "setup.py"' in result.stderr
+    assert "Dev install failed, aborting!" in result.stderr
+    assert result.stdout == ""
+    assert result.returncode == 1
+
+
+def test_bad_interpreter(tmp_path: pathlib.Path, tmp_run: _T_TmpRun) -> None:
```

Comment:
```suggestion
@with_riotfile(
    """
from riot import Venv
venv = Venv(
    ...
)
    """
    , filename="bad-riotfile.py",  # Obviously the default is `riotfile.py`
)
def test_bad_interpreter(riotfile: pathlib.Path, tmp_run: _T_TmpRun) -> None:
```

```suggestion
@with_riotfile_fixture("some-file-in-data-dir.py)
def test_bad_interpreter(riotfile: pathlib.Path, tmp_run: _T_TmpRun) -> None:
```


Just writing something down as a trial to see how it felt, not suggesting a change here.